src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -850,5 +852,11 @@ private NeighborsAndRanges(boolean force, Set<InetAddressAndPort> allNeighbors,
             this.allNeighbors = allNeighbors;
             this.commonRanges = commonRanges;
         }
+
+        List<CommonRange> filterCommonRanges()
+        {
+            // filter and only keep the neighbor endpoints that are contained in `allNeighbors` in each commonRange when in force mode.
+            return RepairRunnable.filterCommonRanges(commonRanges, allNeighbors, force);
```

Comment:
What if we just made `filterCommonRanges()` an instance method on `NeighborsAndRanges` that operated directly on its private state and renamed `RepairRunnableTest` to `NeighborsAndRangesTest`?

Reply:
The intention of `NeighborsAndRanges` is to be used as a simple and class-internal data structure. Moving the method here will open up the access (no more internal to the class) and make it a bit more complex.
Meanwhile, I do see the benefit. I will push a commit to see if we like it.